homeassistant/components/media_player/plex.py

```diff
@@ -192,6 +200,19 @@ def update_devices():
             client.set_availability(client.machine_identifier
                                     in available_client_ids)
 
+            if config.get(CONF_REMOVE_UNAVAILABLE_CLIENTS) \
+                    and not client.available:
+                interval = config.get(CONF_CLIENT_REMOVE_INTERVAL)
+                diff = dt.now() - client.marked_unavailable
+                if diff.total_seconds() >= interval.total_seconds():
```

Comment:
I think that checking it here is weird. Instead, I would expect the logic inside `set_availability`:

```python
if available == self._available:
  return

self._available = True

if available:
  if self._unsub_available:
    self._unsub_available()
    self._unsub_available = None
  else: 
    self._unsub_available = hass.helpers.event.async_call_later(X, self.async_remove)
```

And leverage the callback `async_will_remove_from_hass` on entities to remove it from your known plex clients.

Reply:
I'm not familiar with async and its implementation.. I'll do this

Reply:
If `set_availability()` is called from a sync context, just replace `async_call_later` with `call_later` and you don't do anything async (you might have to create `helpers.event.call_later`, see that file for examples.